python/lsst/daf/butler/script/queryDatasets.py

```diff
@@ -261,7 +262,7 @@ def getDatasets(self) -> Iterator[list[DatasetRef]]:
             summary_datasets=dataset_types,
         )
         expanded_query_collections = [c.name for c in query_collections_info]
-        if self._find_first and set(query_collections) != set(expanded_query_collections):
+        if self._find_first and any(has_globs(c) for c in query_collections):
```

Comment:
```suggestion
        if self._find_first and has_globs(query_collections):
```
has_globs takes an iterable.